resources/function_help/json/to_date

```diff
@@ -1,8 +1,13 @@
 {
   "name": "to_date",
   "type": "function",
-  "description": "Convert a string into a date object.",
-  "arguments": [ {"arg":"string","description":"string representing a date value"}],
-  "examples": [ { "expression":"to_date('2012-05-04')", "returns":"2012-05-04"}
+  "description": "Convert a string into a date object. An optional format string can be provided to parse the string; see QDate::fromString for additional documentation on the format.",
```

Comment:
"see QDate::fromString" -- that's kinda... useless... advice for most of our users :P Could we either link to the Qt docs directly here, or (Better) link to a similar page in the QGIS docs?

Reply:
@nyalldawson , done, I've also taken the opportunity to update the format_date function help hyperlink pointing to an old 4.8 doc link.